### PR TITLE
[Guessing Game] format viewing key error messages

### PR DESCRIPTION
**Pull request by Jennievon**

### CODE REVIEW

**Review by 0xjba — APPROVED** [review]
LGTM

File: guessing-game-v2/src/lib/utils.js [hunk 1]
```diff
@@ -1,6 +1,28 @@
...
   if (!window.gtag) {
     return
   }
   window.gtag('event', eventName, eventData)
 }
+
+export function handleMetaMaskError(error) {
+  if (!error.code) {
+    return error.message
+  }
```
Comment (otherview) [inline guessing-game-v2/src/lib/utils.js:13]: Potential nil return here. Is the error.message guaranteed ? if so can a comment be added to make it explicit ?

File: guessing-game-v2/src/lib/utils.js [hunk 2]
```diff
@@ -1,6 +1,28 @@
...
+  const errorMessage =
+    serializedError?.data?.cause?.error?.data?.message ?? serializedError?.data?.message
+
+  if (errorMessage?.includes('no signed viewing keys')) {
+    return `Missing viewing key. It looks like you may not have registered through the <a href="https://testnet.obscu.ro/" class="connect-link" target="_blank" rel="noopener noreferrer">gateway</a>`
+  }
+
+  if (errorMessage?.includes('invalid viewing key signature for requested address')) {
+    return `Invalid viewing key. Please ensure the connected account is also authenticated via the gateway <a href="https://testnet.obscu.ro/" class="connect-link" target="_blank" rel="noopener noreferrer">here</a>`
+  }
```
Comment (otherview) [inline guessing-game-v2/src/lib/utils.js:25]: I really like this, also think we should have a ticket to make these errors surface in a more meaningful way from the gateway. If we don't have one already :)

**Review by otherview — APPROVED** [review]
lgtm - Lets avoid making formating and feature changes in the same PR please. 
Also is the .lint file updated with the new rules ?

**Pull request merged**